lib/datadog/appsec/event.rb

```diff
@@ -110,12 +114,30 @@ def self.build_service_entry_tags(event_group)
           tags['_dd.appsec.triggers'] += waf_result.events
 
           waf_result.derivatives.each do |key, value|
-            tags[key] = JSON.dump(value)
+            data = Base64.encode64(gzip(JSON.dump(value)))
+
+            if data.size >= MAX_ENCODED_SCHEMA_SIZE
+              Datadog.logger.debug do
+                "Schema key: #{key} exceed max size value. We do not include it as part of the span tags"
+              end
+              next
+            end
+            tags[key] = data
           end
 
           tags
         end
       end
+
+      def self.gzip(value)
```

Comment:
I think there is a decent chance that at some point, somewhere, file IO is going to fail. Should we be explicitly handling exceptions in this method?